public/components/trace_analytics/requests/request_handler.ts

```diff
@@ -35,6 +36,50 @@ export async function handleDslRequest(
   const query = {
     dataSourceMDSId,
   };
+
+  const handleError = (error: any) => {
+    let parsedError = {};
+
+    try {
+      if (typeof error?.response === 'string') {
+        parsedError = JSON.parse(error.response);
+      } else if (typeof error?.body === 'string') {
+        parsedError = JSON.parse(error.body);
+      } else {
+        parsedError = error?.body || error;
+
+        // Check if message is a JSON string
+        if (typeof parsedError.message === 'string') {
+          try {
+            const innerParsed = JSON.parse(parsedError.message);
+            if (innerParsed?.error) {
+              parsedError = innerParsed;
+            }
+          } catch {
+            // not JSON, skip
+          }
```

Comment:
Should this be empty?

Reply:
I think we don't a separate try catch here.

Reply:
Removed empty catch